File: app/model/SyncLog.js
// app/model/SyncLog.js
import { Model } from '@nozbe/watermelondb';
import { field, text, date, json } from '@nozbe/watermelondb/decorators';

export default class SyncLog extends Model {
  static table = 'sync_logs';

  @field('device_id') deviceId;
  @field('entity_type') entityType;
  @field('entity_id') entityId;
  @field('operation') operation;
  @json('payload', (json) => json) payload;
  @date('client_timestamp') clientTimestamp;
  @date('server_timestamp') serverTimestamp;
  @field('conflict_resolved') conflictResolved;
}

File: src/main/java/com/carenode/entity/SyncLog.java
package com.carenode.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "sync_log")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SyncLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String deviceId;

    private String entityType;

    private java.util.UUID entityId;

    private String operation;

    @Column(columnDefinition = "jsonb")
    private String payload;

    private LocalDateTime clientTimestamp;

    private LocalDateTime serverTimestamp;

    private Boolean conflictResolved = false;
}